shell/common/rasterizer.cc

```diff
@@ -158,12 +158,9 @@ void Rasterizer::DrawLastLayerTree(
   }
 }
 
-RasterStatus Rasterizer::Draw(
-    std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder,
-    std::shared_ptr<Pipeline<flutter::LayerTree>> pipeline,
-    LayerTreeDiscardCallback discard_callback) {
-  TRACE_EVENT_WITH_FRAME_NUMBER(frame_timings_recorder, "flutter",
-                                "GPURasterizer::Draw");
+RasterStatus Rasterizer::Draw(std::shared_ptr<LayerTreePipeline> pipeline,
+                              LayerTreeDiscardCallback discard_callback) {
+  TRACE_EVENT0("flutter", "GPURasterizer::Draw");
```

Comment:
Since `frame_timings_recorder` is in the pipeline, it can't be accessed here. So I changed the code here back to before this PR https://github.com/flutter/engine/pull/26205 and added a trace in `DoDraw`. It looks like this will have an impact on the dev tools. Please let me know if you guys have any thoughts.

Reply:
This might impact the devtools, as it looks for specific frame number events corresponding to `::Draw`. @kenzieschmoll  do we still rely on the names of trace events or do we only care about the frame timings API?

Reply:
We look for the 'frame_number' arg in the 'GPURasterizer::Draw' trace event and the 'Animator::BeginFrame' trace event to link timeline events to their matching frame number from the FrameTimings API. If this change removes the 'frame_number' argument, this will break DevTools.

Reply:
I filed https://github.com/flutter/flutter/issues/105140, once that is addressed, we can land this PR.